### ISSUE TITLE
CDATA literal text error due to Data Binding, and newlines lost

### ISSUE THREAD

**Issue by estanglerbm** [apache/royale-compiler#213]
When using CDATA for literals that contain something that looks like data binding (i.e. {sum}), getting error:

```
Warning: Data binding will not be able to detect assignments to 'sum'.
```

Also (if you replace the problematic part with normal text), newlines are being lost (Flex preserves them).

[TestLiteralsFlex.mxml.txt](https://github.com/apache/royale-compiler/files/8721450/TestLiteralsFlex.mxml.txt)
[TestLiterals.mxml.txt](https://github.com/apache/royale-compiler/files/8721451/TestLiterals.mxml.txt)

**Comment by joshtynjala** [comment]
I fixed the compiler issue where it incorrectly detected `{` and `}` as data binding inside CDATA. The missing new lines are because the MX `TextArea` component in Royale has `[CollapseWhiteSpace]` metadata on the `text` property. If you think that's wrong, it'll need to be handled on the royale-asjs side. The compiler is handling `[CollapseWhiteSpace]` correctly.

https://github.com/apache/royale-asjs/blob/apache-royale-0.9.9/frameworks/projects/MXRoyale/src/main/royale/mx/controls/TextArea.as#L1697

**Comment by joshtynjala** [comment]
Hmmm... actually, MX `TextArea` in the Flex SDK also has `[CollapseWhiteSpace]`.

https://github.com/apache/flex-sdk/blob/apache-flex-sdk-4.16.1/frameworks/projects/mx/src/mx/controls/TextArea.as#L1733

So it looks like whitespace is collapsed _except_ when it is contained within CData.
